pkg/util/funcutil/func.go

```diff
@@ -49,6 +49,27 @@ func CheckGrpcReady(ctx context.Context, targetCh chan error) {
 	}
 }
 
+// GetIP return the ip address
+func GetIP(ip string) string {
+	if len(ip) == 0 {
+		return GetLocalIP()
+	}
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		panic(err)
+	}
+	if err == nil {
+		for _, addr := range addrs {
+			ipaddr, ok := addr.(*net.IPNet)
+			if ipaddr.IP.String() == ip && ok && ipaddr.IP.IsGlobalUnicast() && ipaddr.IP.To4() != nil {
+				return ipaddr.IP.String()
+			}
+		}
+	}
+	panic("specified ip is invalid")
```

Comment:
Why did copy the code for GetLocalIP again? 🤔

Reply:
here has check the incoming ip is one of the local ip and is valid, GetLocalIP just return the first valid ip.

Reply:
I don't think it's necessary to check. On one hand, the port binding in grpc will perform a check, and on the other hand, we haven't considered the case of IPv6 here. 🤔